en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_jQuery_Entwine.md

```diff
@@ -0,0 +1,491 @@
+
+Given this DOM structure:
+
+```html
+<body>
+  <div class="internal_text">Internal text</div>
+  <div class="attribute_text" rel="Attribute text"></div>
+  <div>Nonsense</div>
+</body>
+```
+
+And this entwine definition
+
+```js
+$('.internal_text').entwine({
+  foo: function() {
+    console.log(this.text());
+  },
+});
+
+$('.attribute_text').entwine({
+  foo: function() {
+    console.log(this.attr('rel'));
+  },
+});
+```
+
+Then this call
+
+```js
+$('div').foo();
+```
+
+Will log this to the console
+
+```
+Internal text
+Attribute text
+```
+
+Note that it is calling the `foo()` method on _both_ divs, and that each had a different `foo()` method defined based on different selectors.
+
+## Events
+
+If you declare a function with a name starting with 'on', then instead of defining that function as a callable method, it will be bound to an event of that
+name. Just like other functions this binding will be live, and only the most specific definition will be used.
+
+```html
+<div>Background will turn blue when clicked on</div>
+<div>Will also have blue background when clicked on</div>
+<div class='green'>Will have green text when clicked on. Background colour will not change</div>
+```
+
+```js
+/* No need for onready wrapper. Events are bound as needed */
+$('div').entwine({
+  onclick: function() {
+    this.css({backgroundColor: 'blue'});
+  },
+});
+
+$('.green').entwine({
+  onclick: function() {
+    this.css({color: 'green'});
+  },
+});
+```
+
+[hint]
+Remember, if you wanted the background colour to change for the div with class `green` as well, you can simply call `this._super()` in the click event handler declared for that selector. See [Selector specifity and "inheritance"](#specificity) for more information about how this works.
+[/hint]
+
+### Handling events from other elements
+
+Sometimes we want one element to react to events that are occuring on another element. For these situations, the special `from` syntax can be used. This should be used sparingly.
+
+Examples of where this can be useful are if the logic for the element the events are happening on is declared in a different file to the logic for the element you want to perform actions on, or if you have extensive API declared for the element you want to perform actions on that would be cumbersome to call from the element that owns the event.
+
+```html
+<a>The click event for this element will be handled by the div!</a>
+<div class='green'>Will have green text when the link is clicked on.</div>
+```
+
+```js
+$('div').entwine({
+  'from a': {
+    onclick: function() {
+      this.css({color: 'green'});
+      this._super();
+    },
+  },
+});
+```
+
+## Constructors / Destructors
+
+Declaring a function with the name `onmatch` will create a behavior that is called on each object when it matches. Likewise, `onunmatch` will
+be called when an object that did match this selector stops matching it (because it is removed, or because you've changed its properties).
```

Comment:
```suggestion
Declaring a function with the name `onmatch` will create a behavior that is called on each object when it matches. Likewise, `onunmatch` will be called when an object that did match this selector stops matching it (because it is removed, or because you've changed its properties).
```

Reply:
I don't understand the need for this change, but it won't change anything so I'll do it lol

Reply:
Done